pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -3759,3 +3305,112 @@ func getTaskRunStatus(t string, status corev1.ConditionStatus) *v1beta1.Pipeline
 		},
 	}
 }
+
+func buildTestData(
+	prs []*v1beta1.PipelineRun,
+	ps []*v1beta1.Pipeline,
+	ts []*v1beta1.Task,
+	clusterTasks []*v1beta1.ClusterTask,
+	trs []*v1beta1.TaskRun,
+	rs []*v1alpha1.PipelineResource,
+	conditions []*v1alpha1.Condition,
+) test.Data {
+	return test.Data{
+		PipelineRuns:      prs,
+		Pipelines:         ps,
+		TaskRuns:          trs,
+		Tasks:             ts,
+		ClusterTasks:      clusterTasks,
+		PipelineResources: rs,
+		Conditions:        conditions,
+	}
+}
+
+func getReconciledRun(
```

Comment:
I think it's important to try to indicate with naming whether mutations / assertions are happening in functions - reading the name of `getReconciledRun` I would expect it to be only retrieving a run, but it seems to be doing several things:

- instantiating a controller
- reconciling
- asserting against events

Some alternative naming ideas: `reconcileRun` (dropping "get" makes it seem less like it's just retrieving) `assertRunSucceeds`

Another idea, mentioned below as well: create a struct that holds all the args below and expose several functions instead of just one - this will probably still be significantly easier to read and work with than the amount of repetition we have today

Reply:
renamed `getReconciledRun` to `reconcileRun`

Creating `struct` is a great suggestion, I created a struct [PipelineRunTest](https://github.com/tektoncd/pipeline/pull/2876/files#diff-f7c8532cfe79319c0fff711a4979a6d6R79) and made `reconcileRun` as an attribute to `PipelineRunTest`.